### PR TITLE
Add list deployment operation command

### PR DESCRIPTION
**Pull request by vivsriaus**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vivsriaus__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (visriniv). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: test/commands/arm/group/arm.group.deployment.operation-tests.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        if (suite.isPlayback()) {
+          setTimeout = function (action, timeout) {
+            process.nextTick(action);
+          };
+        }
+        suite.setupSuite(done);
+      });
+      
+      after(function (done) {
+        if (!suite.isPlayback()) {
```
Comment (yugangw-msft) [inline test/commands/arm/group/arm.group.deployment.operation-tests.js]: should not you do it only under "palyback"?, why the opposite?

File: test/recordings/arm-cli-deployment-operation-tests/suite.arm-cli-deployment-operation-tests.nock.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    registeredProviders: [],
+    isDefault: true
+  }, newProfile.environments['AzureCloud']));
+
+  return newProfile;
+};
+
+exports.setEnvironment = function() {
+  process.env['AZURE_ARM_TEST_LOCATION'] = 'West US';
+};
```
Comment (yugangw-msft) [inline test/recordings/arm-cli-deployment-operation-tests/suite.arm-cli-deployment-operation-tests.nock.js]: this record file has nothing inside, expected?

File: lib/commands/arm/group/group.deployment.operation._js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      }
+
+      var subscription = profile.current.getSubscription(options.subscription);
+      var client = utils.createResourceClient(subscription);
+      var operationsList = [];
+
+      cli.interaction.withProgress($('Getting deployoment operations'),
+        function (log, _) {
+          var operationsResult = client.deploymentOperations.list(resourceGroup, name, _);
+          operationsList.push.apply(operationsList, operationsResult.operations);
```
Comment (yugangw-msft) [inline lib/commands/arm/group/group.deployment.operation._js]: why you can't just do "operationsList.push(operationsResult.operations)"? Any JS trick here?

File: lib/commands/arm/group/group.deployment.operation._js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+          while (operationsResult.nextLink) {
+            log.info($('Getting more operations'));
+            operationsResult = client.deploymentOperations.listNext(operationsResult.nextLink, _);
+            operationsList.push.apply(operationsList, operationsResult.operations);
```
Comment (yugangw-msft) [inline lib/commands/arm/group/group.deployment.operation._js]: same here

File: test/commands/arm/group/arm.group.deployment.operation-tests.js (hunk 1 above)
Comment (vivsriaus) [inline test/commands/arm/group/arm.group.deployment.operation-tests.js]: This is the standard we follow for deployment tests as well

File: test/recordings/arm-cli-deployment-operation-tests/suite.arm-cli-deployment-operation-tests.nock.js (hunk 2 above)
Comment (vivsriaus) [inline test/recordings/arm-cli-deployment-operation-tests/suite.arm-cli-deployment-operation-tests.nock.js]: Yes, this was output after recording, and I see similar files across all our tests

File: lib/commands/arm/group/group.deployment.operation._js (hunk 3 above)
Comment (vivsriaus) [inline lib/commands/arm/group/group.deployment.operation._js]: This is merging two arrays, so apply is needed. More here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push

File: lib/commands/arm/group/group.deployment.operation._js (hunk 4 above)
Comment (vivsriaus) [inline lib/commands/arm/group/group.deployment.operation._js]: Same as above

File: lib/commands/arm/group/group.deployment.operation._js (hunk 3 above)
Comment (yugangw-msft) [inline lib/commands/arm/group/group.deployment.operation._js]: i see.

**Pull request merged**